Mathlib/RingTheory/Ideal/Quotient.lean

```diff
@@ -348,4 +347,13 @@ theorem map_pi {ι : Type*} [Finite ι] {ι' : Type w} (x : ι → R) (hi : ∀
 
 end Pi
 
+/-- A ring is made up of a disjoint union of cosets of an image. -/
+lemma univ_eq_iUnion_image_add {R : Type*} [Ring R] (I : Ideal R) :
+    (Set.univ (α := R)) = ⋃ x : R ⧸ I, (x.out' + ·) '' I :=
+  QuotientAddGroup.univ_eq_iUnion_image_add I.toAddSubgroup
```

Comment:
I'm not sure if this is useful. `.out'` is quite hard to work with.

Reply:
One can use `Ideal.Quotient.eq` to relate to `out'`. I'll link how I use it in the larger proof.

Reply:
https://github.com/leanprover-community/mathlib4/pull/15424/files#diff-e2953739e89168dcbb0040fe18b7996a54734ee000cfa73813c6a324d3c2f8a9R186-R192

This way, it discusses balls around `i.out` and `y.out`, because it's an easy way to talk about representatives of the residue field (or other quotients) inside the ring itself.